server/game/core/ability/abilityTargets/CardTargetResolver.ts

```diff
@@ -17,8 +17,16 @@ export class CardTargetResolver extends TargetResolver<ICardTargetResolver<Abili
     private selector;
     public constructor(name: string, properties: ICardTargetResolver<AbilityContext>, ability: PlayerOrCardAbility) {
         super(name, properties, ability);
+
         this.selector = this.getSelector(properties);
 
+        // a player can always choose not to pick a card from a zone that is hidden from their opponents (SWU Comp Rules 2.0 1.17.4)
+        if (this.allZonesAreHidden(this.properties.locationFilter)) {
+            this.properties.optional = true;
+            this.selector.optional = true;
+            this.selector.oldDefaultActivePromptTitle = this.selector.defaultActivePromptTitle;
```

Comment:
Is this needed?

Reply:
I think so. The different card selectors have different default prompts(`'Choose a ' + this.cardTypeFilter[0];`, `'Choose cards'`, ``Select ${this.numCards} cards``) so if we want a note for the player about why they can choose no targets like you suggested I think it makes sense to just append it to the end. Is there a better way of doing that?

Reply:
Ahh sorry, didn't realize these were functions not properties. Makes more sense now. Yeah it's annoying that they're all different, would be much easier if they had a base class so we can override the functionality in one place. Normally I wouldn't want to silently override a function's behavior like this but it's a pretty innocuous place to do it so I think it's fine as a band-aid.